README.md

```diff
@@ -171,11 +171,36 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
     *   By default, the worklet uses the invoking context's origin as its partition origin for accessing shared storage data and for budget checking and withdrawing.
         *   To instead use the worklet script origin (i.e. `url`'s origin) as the partition origin for accessing shared storage, pass the `dataOrigin` option with "script-origin" as its value in the `options` dictionary.
-        *   Currently, the `dataOrigin` option, if used, is restricted to having either "script-origin" or "context-origin" as its value. "script-origin" designates the worklet script origin as the data partition origin; "context-origin" designates the invoking context origin as the data partition origin.
+        *   To use a custom origin as the partition origin for accessing shared storage, pass the `dataOrigin` option with the serialized partition origin (i.e. the [URL serialization](https://url.spec.whatwg.org/#url-serializing) of a URL with the same scheme, host, and port as the partition origin, but whose other components are empty) as its value in the `options` dictionary.
+        *   Supported values for the `dataOrigin` option, if used, are the keywords "script-origin" and "context-origin", as well as any valid serialized HTTPS origin, e.g. "https://custom-data-origin.example". 
+            *   "script-origin" designates the worklet script origin as the data partition origin.
+            *   "context-origin" (default) designates the invoking context origin as the data partition origin.
+            *   If a valid serialized HTTPS URL is passed as the value for `dataOrigin`, then the origin of the URL parsed from the `dataOrigin` string will be used as the worklet's data partition origin, as long as the URL's origin has opted-in via a [/.well-known/](#well-known) to specify that the worklet script's origin and context origin are trusted.
     *   The object that the returned Promise resolves to has the same type with the implicitly constructed `window.sharedStorage.worklet`. However, for a worklet created via `window.sharedStorage.createWorklet(url, options)`, only `selectURL()` and `run()` are available, whereas calling `addModule()` will throw an error. This is to prevent leaking shared storage data via `addModule()`, similar to the reason why `addModule()` can only be invoked once on the implicitly constructed `window.sharedStorage.worklet`.
     *   Redirects are not allowed.
-    *   When the module script's URL's origin is cross-origin with the worklet's creator window's origin and when `dataOrigin` is "script-origin", a `Shared-Storage-Cross-Origin-Worklet-Allowed: ?1` response header is required.
+    *   When the module script's URL's origin is cross-origin with the worklet's creator window's origin and when `dataOrigin` is "script-origin" (or is a valid serialized HTTPS URL that is same-origin to the worklet's script's origin), a `Shared-Storage-Cross-Origin-Worklet-Allowed: ?1` response header is required.
```

Comment:
Do we still require the ./well-known mechanism in this case? I think either one of the following is reasonable:  1) we skip the ./well-known check completely, and just rely on the script's response header. 2) if either mechanism indicates "allowed", then we'll allow it.

In either case, let's clearly document the mechanism.

Reply:
I was thinking we do (1), skip the /.well-known/ check completely, and just rely on the script's response header.

I will make that explicit.

Reply:
Done.